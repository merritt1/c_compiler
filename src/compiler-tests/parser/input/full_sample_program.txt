unsigned int abc;
signed char a;
char b;
int c;
int a, b, c;
char d[3], *e, **f;
int a[1][2][3];
void fn(int, char, int);
int fn1(char[][2]);
char fn3(char*, int*);
int a(int b, int c);

int main(void) {
	a=b=c;
	c++;
	a = d-- + c / 3 % 4;
	if (a || b || c && d) {
		goto label;
		return;
		break;
		continue;
	} else {
		a++;
	}

	while (a>0) {
		b--;
	}
}

char test(int a, char b) {
	main(123);
}
