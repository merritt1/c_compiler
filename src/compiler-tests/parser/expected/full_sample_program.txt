unsigned int (abc);
signed char (a);
signed char (b);
signed int (c);
signed int (a), (b), (c);
signed char (d[3]), (*(e)), (**(f));
signed int (a[1][2][3]);
void (fn(signed int, signed char, signed int));
signed int (fn1(signed char ([][2])));
signed char (fn3(signed char (*), signed int (*)));
signed int (a(signed int (b), signed int (c)));
signed int (main(void)) {
(a=(b=c));
(c++);
(a=((d--)+((c/3)%4)));
if (((a||b)||(c&&d))) {
goto label;
return;
break;
continue;
}
else {
(a++);
}
while ((a>0)) {
(b--);
}
}
signed char (test(signed int (a), signed char (b))) {
(main(123));
}
